server/src/services/person.service.ts

```diff
@@ -376,13 +378,26 @@ export class PersonService {
     return JobStatus.SUCCESS;
   }
 
-  async handleQueueRecognizeFaces({ force }: IBaseJob): Promise<JobStatus> {
+  async handleQueueRecognizeFaces({ force, nightly }: INightlyJob): Promise<JobStatus> {
     const { machineLearning } = await this.configCore.getConfig({ withCache: false });
     if (!isFacialRecognitionEnabled(machineLearning)) {
       return JobStatus.SKIPPED;
     }
 
     await this.jobRepository.waitForQueueCompletion(QueueName.THUMBNAIL_GENERATION, QueueName.FACE_DETECTION);
+
+    if (nightly) {
+      const [state, latestFaceDate] = await Promise.all([
+        this.systemMetadataRepository.get(SystemMetadataKey.FACIAL_RECOGNITION_STATE),
+        this.repository.getLatestFaceDate(),
+      ]);
+
+      if (state?.lastRun && latestFaceDate && state.lastRun > latestFaceDate) {
```

Comment:
Is this a string compare? Does that even work?

Reply:
Yes, since they're both ISO strings.

Reply:
I tested by hardcoding nightly: true. The first time it ran (since there was no system metadata entry) and the second time it skipped.

Reply:
Ok. Maybe we should add a comment or something. I think it isn't very obvious, which makes it an easy place to introduce a bug.